### ISSUE TITLE
Amplify push always shows update under Api Operation

### ISSUE THREAD

**Issue by BatteryAcid** [aws-amplify/amplify-category-api#738]
> **Note**: If your question is regarding the AWS Amplify Console service, please log it in the
> [AWS Amplify Console repository](https://github.com/aws-amplify/amplify-console/issues)

**Which Category is your question related to?**
Amplify CLI

**Amplify CLI Version**
You can use `amplify -v` to check the amplify cli version on your system
9.2.1

**What AWS Services are you utilizing?**
Lambda

**Provide additional details e.g. code snippets. _Be sure to remove any sensitive data._**
Running amplify status doesn't show any changes, but amplify push always shows my api has an update:

<img width="634" alt="Screen Shot 2022-08-16 at 12 25 05 PM" src="https://user-images.githubusercontent.com/8283698/184930373-65cf195a-f780-4ce5-9ffb-0254d4cb7fc8.png">

Is there a way to reset it?  I've ran amplify pull and then a push, but it still keeps coming back with changes detected.
I've even made a change to the schema to try and force it, but no luck.

Thanks

**Comment by josefaidt** [comment]
Hey @BatteryAcid :wave: thanks for raising this! To better understand the context of the issue, does this operation status reset to "No change" after `amplify push` or is this something you see not changing regardless of push? Or perhaps is the `amplify pull` causing the API to then show as UPDATE?

As a follow-up, do you have any functions that are attached to this API via `@function` or a function that has permissions granted to the API?

**Comment by BatteryAcid** [comment]
It's something I see not changing regardless of push. I can run amplify push after making changes or not, and it's still there.  Amplify pull doesn't seem to have an effect.

I have one `@function` inside a `mutation {}` block in my schema.graphql.

**Comment by josefaidt** [comment]
Hey @BatteryAcid :wave: apologies for the delay here!! Does `amplify pull --restore` have an effect on this UPDATE status? Are you using any overrides with this resource?

**Comment by BatteryAcid** [comment]
So, I started getting getting an API Key not found error and I believe I resolved it using a mix of the following solutions (see below), not exactly sure here, it was pretty muddy.  But at the end of it I'm not seeing it stuck in update anymore.  I will try the --restore option if I run into this again.  Thanks for the response.

- https://stackoverflow.com/a/71306889/1956540
- https://github.com/aws-amplify/amplify-cli/issues/808#issuecomment-596217380
